### DISCUSSION TITLE
Preapprove app access or alternative to OAUTH redirect_url?

### DISCUSSION THREAD

**Discussion by littlebee** [etsy/open-api#1119] [Q&A] [answered]
Hi everyone. 👋 

We're a small Mom and Pop shop (coming soon!) where Mom and Pop happen to be retired software engineers.  So of course we have all of our product data in our own in-house database and complementary NextJS app and gql interfaces, backed up in the cloud, ....

What I need to do now is get that data into our shop on Etsy.   I immediately reached for the API, but it seems to require a public-facing, SSL cert encrypted web front-end to send the OAuth code via the provided `redirect_url`.   This is only for our shop and not something I intend to release or open source.  My public-facing URI is simply a redirect to the Etsy shop.  

Is there another way to sync my in-house (literally 😆) system with my Etsy shop?   Or maybe a way to pre-authorize my app to do crud on my shop?   I really just don't want to manage a public-facing system, pay for certs, etc. quite yet.

Thank you for reading this far :) 

-bee

**Comment by turboturtleD** [comment, marked as answer]
I'm sure someone will correct me if I'm wrong, but I believe some other folks have used ngrok to validate using a localhost.  You do have to use OAuth2 for validation, there is no workaround there.  Alternately, if you do setup a webserver you can of course use "Let's Encrypt" to get free SSL certs.

**Reply by littlebee** [reply]
Thank you!   Now if I can just get through the Etsy api approval process :/

**Comment by thisisjake** [comment]
The redirct_url is only used during the initial authentication, previously I used Postman for that, and once I had the refresh token (and assuming I don't let it expire), I had no need to authenticate again as I could keep using the token.

Alternatively you could do what @turboturtleD stated.

EDIT: Take a look at this:
(https://stackoverflow.com/questions/68353440/making-etsy-oauth2-0-call-via-postman)
